functests/1_performance/rt-kernel.go

```diff
@@ -30,33 +32,43 @@ var _ = Describe("[performance]RT Kernel", func() {
 
 	It("[test_id:26861][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] should have RT kernel enabled", func() {
 
-		Eventually(func() string {
+		profile := &performancev1alpha1.PerformanceProfile{}
+		key := types.NamespacedName{
+			Name: testutils.PerformanceProfileName,
+		}
+		err := testclient.Client.Get(context.TODO(), key, profile)
+		Expect(err).ToNot(HaveOccurred(), "failed to get the performance profile")
+
+		if profile.Spec.RealTimeKernel != nil && *profile.Spec.RealTimeKernel.Enabled == true {
 
-			// run uname -a in a busybox pod and get logs
-			testpod = pods.GetTestPod()
-			testpod.Namespace = testutils.NamespaceTesting
-			testpod.Spec.Containers[0].Command = []string{"uname", "-a"}
-			testpod.Spec.RestartPolicy = corev1.RestartPolicyNever
-			testpod.Spec.NodeSelector = map[string]string{
-				fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerCNF): "",
-			}
+			Eventually(func() string {
 
-			if err := testclient.Client.Create(context.TODO(), testpod); err != nil {
-				return ""
-			}
+				// run uname -a in a busybox pod and get logs
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Command = []string{"uname", "-a"}
+				testpod.Spec.RestartPolicy = corev1.RestartPolicyNever
+				testpod.Spec.NodeSelector = map[string]string{
+					fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerCNF): "",
+				}
 
-			if err := pods.WaitForPhase(testpod, corev1.PodSucceeded, 60*time.Second); err != nil {
-				return ""
-			}
+				if err := testclient.Client.Create(context.TODO(), testpod); err != nil {
+					return ""
```

Comment:
Is the returning of an empty string some kind of error flow? I find it odd

Reply:
its tested against the clause in the end of eventually `.Should(ContainSubstring("PREEMPT RT")`
this is existing code

Reply:
thanks

Reply:
So maybe we should use predefined token like return FAILURE (var FAILURE = "") ?

Reply:
can be, but im fine with the current form